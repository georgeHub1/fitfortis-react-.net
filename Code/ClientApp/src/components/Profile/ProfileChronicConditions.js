import React, { Component } from 'react';
import { Form } from 'antd';
import { injectIntl} from 'react-intl';
import PropTypes from 'prop-types';
import CustomTag from '../Common/CustomTag';
import CustomCard from '../Common/CustomCard.js';
import styles from './styles.module.less';
import * as R from 'ramda';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';

class ProfileChronicConditions extends Component {
  static propTypes = {
    setTabValue: PropTypes.func,
    profileChronicConditions: PropTypes.array
  };
  static defaultProps = {
    profileChronicConditions: [],
    setTabValue: () => true
  };
  constructor (props) {
    super(props);
    this.state = {
      profileChronicConditions: props.profileChronicConditions
    };
  }

  componentDidMount () {
    analyticId.firebaseAnalyticsLog('Profile_ChronicConditions');
  }
  componentDidUpdate ({profileChronicConditions}) {
    const { profileChronicConditions: profileChronicConditionsProps} = this.props;

    if (!R.equals(profileChronicConditions, profileChronicConditionsProps))
    this.setState({
      profileChronicConditions: [...profileChronicConditionsProps]
    });
  }

  onClickTag = (tag, key) => {
    const { setTabValue } = this.props;
    const { profileChronicConditions } = this.state;

    const data = R.clone(profileChronicConditions);
    const indexElem = R.indexOf(tag)(data);
    const newElem = R.assoc('isSelected', !tag.isSelected)(tag);

    data.splice(indexElem, 1, newElem);
    setTabValue({
      profileChronicConditions: [
        ...data
      ]
    });
    this.setState({
      profileChronicConditions: [
        ...data
      ]
    });
  };

  render () {
    const profileChronicConditions = this.state.profileChronicConditions !== undefined
      ? R.groupWith((a, b) => a.key === b.key, R.sortBy(R.prop('key'), R.clone(this.state.profileChronicConditions)))
      : [];
    const {genderIdentity} = this.props;

    return (
      <div>
      {
        profileChronicConditions.map((groupChronicCondition, index) => {
          return (
            (genderIdentity === 1 && groupChronicCondition[0].key === 'Gynecologic')
            ? null
            : <CustomCard key ={index}>
                <div className={styles.chronicConditionsWrapper}>
                  <div className="cardHeaderProfile">
                  { groupChronicCondition[0].category }
                  </div>
                  <div className="containerTags">
                    <Form.Item>
                      <CustomTag
                        handleClick={this.onClickTag}
                        tags={ groupChronicCondition }
                        keyTag='profileChronicConditions'
                      />
                    </Form.Item>
                  </div>
                </div>
              </CustomCard>
          );
        })
      }
      </div>
    );
  }
}

export default injectIntl(ProfileChronicConditions);
