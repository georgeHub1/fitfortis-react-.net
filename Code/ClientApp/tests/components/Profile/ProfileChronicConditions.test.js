import React from 'react';
import renderer from 'react-test-renderer';

import ProfileChronicConditions from '../../../src/components/Profile/ProfileChronicConditions.js';

const Container = ProfileChronicConditions;

describe('ProfileChronicConditions', () => {
  it('renders correctly', () => {
    const wrapper = renderer
      .create(<Container/>)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
