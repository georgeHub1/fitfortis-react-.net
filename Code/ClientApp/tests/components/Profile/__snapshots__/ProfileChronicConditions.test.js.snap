// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileChronicConditions renders correctly 1`] = `<div />`;
